validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -601,3 +601,54 @@ def _respond(self, request):
             LOGGER.debug(e)
             return self._status.NO_RESOURCE
         return self._wrap_response(batch=batch)
+
+
+class TransactionListRequest(_ClientRequestHandler):
+    def __init__(self, block_store):
+        super().__init__(
+            client_pb2.ClientTransactionListRequest,
+            client_pb2.ClientTransactionListResponse,
+            validator_pb2.Message.CLIENT_TRANSACTION_LIST_RESPONSE,
+            block_store=block_store)
+
+    def _respond(self, request):
+        head_id = self._get_head_block(request).header_signature
+        transactions = self._list_store_resources(
+            request,
+            head_id,
+            request.transaction_ids,
+            self._block_store.get_transaction,
+            lambda block: [t for a in block.batches for t in a.transactions])
+
+        transactions, paging = _Pager.paginate_resources(
+            request,
+            transactions,
+            self._status.INVALID_PAGING)
+
+        if not transactions:
```

Comment:
Because of the way Protobufs work there is no way to tell the difference between an empty list of data and data that wasn't set at all. For that reason, when there is no data the List handlers explicitly return a NO_RESOURCE status.